Panic when map range loop and return error in function

```go
func func1() error {
	maps := make(map[string]string)

	for range maps {
	}

	return nil
}
```

```
panic: unreachable

goroutine 1202 [running]:
go/types.unreachable(...)
        /usr/local/Cellar/go/1.11.5/libexec/src/go/types/errors.go:23
go/types.identical(0x1449100, 0x169bd20, 0x1448e80, 0xc00008cb40, 0xc000897801, 0x0, 0xc000897800)
        /usr/local/Cellar/go/1.11.5/libexec/src/go/types/predicates.go:292 +0xd44
go/types.Identical(0x1449100, 0x169bd20, 0x1448e80, 0xc00008cb40, 0xc00008caf0)
        /usr/local/Cellar/go/1.11.5/libexec/src/go/types/predicates.go:116 +0x57
go/types.(*operand).assignableTo(0xc0006bec20, 0x0, 0x1448e80, 0xc00008cb40, 0x0, 0x0)
        /usr/local/Cellar/go/1.11.5/libexec/src/go/types/operand.go:212 +0xb1
go/types.AssignableTo(0x1449100, 0x169bd20, 0x1448e80, 0xc00008cb40, 0x1d080d8)
        /usr/local/Cellar/go/1.11.5/libexec/src/go/types/api.go:363 +0x91
github.com/sachaos/xerrchk/passes/wrapping.isErr(0x1449100, 0x169bd20, 0x169bd20)
        /Users/sonatard/go/src/github.com/sachaos/xerrchk/passes/wrapping/wrapping.go:155 +0x4d
github.com/sachaos/xerrchk/passes/wrapping.wrappingErrPositions(0xc000d00140, 0x0, 0x0, 0x0)
        /Users/sonatard/go/src/github.com/sachaos/xerrchk/passes/wrapping/wrapping.go:78 +0x1af
github.com/sachaos/xerrchk/passes/wrapping.run(0xc000c56750, 0xc0008ecfc0, 0x16a06e0, 0xc0006a1b18, 0xc000142058)
        /Users/sonatard/go/src/github.com/sachaos/xerrchk/passes/wrapping/wrapping.go:45 +0x118
golang.org/x/tools/go/analysis/internal/checker.(*action).execOnce(0xc00014f720)
        /Users/sonatard/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:513 +0x5f7
golang.org/x/tools/go/analysis/internal/checker.(*action).execOnce-fm()
        /Users/sonatard/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:434 +0x2a
sync.(*Once).Do(0xc00014f720, 0xc0001cb798)
        /usr/local/Cellar/go/1.11.5/libexec/src/sync/once.go:44 +0xb3
golang.org/x/tools/go/analysis/internal/checker.(*action).exec(0xc00014f720)
        /Users/sonatard/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:434 +0x50
golang.org/x/tools/go/analysis/internal/checker.execAll.func1(0xc00014f720)
        /Users/sonatard/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:422 +0x34
created by golang.org/x/tools/go/analysis/internal/checker.execAll
        /Users/sonatard/go/src/golang.org/x/tools/go/analysis/internal/checker/checker.go:428 +0x100
```

--- a/passes/wrapping/testdata/src/a/map_range.go
+++ b/passes/wrapping/testdata/src/a/map_range.go
@@ -1,0 +1,10 @@
+package a
+
+func MapRange() error {
+	maps := make(map[string]string)
+
+	for range maps {
+	}
+
+	return nil
+}

--- a/passes/wrapping/testdata/src/a/map_range.go
+++ b/passes/wrapping/testdata/src/a/map_range.go
@@ -1,5 +1,6 @@
 package a
 
+// NOTE: Case https://github.com/sachaos/xerrchk/issues/6
 func MapRange() error {
 	maps := make(map[string]string)
 

--- a/passes/wrapping/wrapping.go
+++ b/passes/wrapping/wrapping.go
@@ -42,7 +42,7 @@ func run(pass *analysis.Pass) (interface{}, error) {
 
 	srcFuncs := pass.ResultOf[buildssa.Analyzer].(*buildssa.SSA).SrcFuncs
 	for _, srcFunc := range srcFuncs {
-		positions := wrappingErrPositions(srcFunc)
+		positions := wrappingErrPositions(pass, srcFunc)
 		for _, pos := range positions {
 			pass.Reportf(pos, "wrap with xerrros.Errorf or xerrors.Opaque")
 		}
@@ -58,7 +58,7 @@ func checkFlags() error {
 	return fmt.Errorf("unknown scope '%s'", scope)
 }
 
-func wrappingErrPositions(srcFunc *ssa.Function) []token.Pos {
+func wrappingErrPositions(pass *analysis.Pass, srcFunc *ssa.Function) []token.Pos {
 	if scope == scopePublic && isPrivate(srcFunc) {
 		return nil
 	}
@@ -75,13 +75,18 @@ func wrappingErrPositions(srcFunc *ssa.Function) []token.Pos {
 				continue
 			}
 
-			if isErr(val.Type()) && !isCallingXerrors(val) && isReachToReturn(val) {
+			if !isRange(val) && isErr(val.Type()) && !isCallingXerrors(val) && isReachToReturn(val) {
 				positions = append(positions, convertToOriginVal(val).Pos())
 			}
 		}
 	}
 
 	return positions
+}
+
+func isRange(val ssa.Value) bool {
+	_, ok := val.(*ssa.Range)
+	return ok
 }
 
 func isPrivate(function *ssa.Function) bool {

